internal/co_coders/criteria/coding_language.go

```diff
@@ -1,60 +1,12 @@
 package criteria
 
-// CodingLanguage ..
-type CodingLanguage int
 
-// CodingLanguage constants
-const (
-	AssemblyLanguage CodingLanguage = iota
-	Bash
-	C 
-	Clojure
-	CPP
-	CSharp
-	Dart
-	Erlang
-	FSharp
-	Go
-	Haskell
-	Java
-	JavaScript
-	Kotlin
-	Lisp
-	ObjectiveC
-	ObjectPascal
-	Perl
-	PHP
-	Python
-	R
-	Ruby
-	Rust
-	Scala
-	Swift
-	SQL
-	TypeScript
-	VisualBasic
-)
+// NewCodingLanguage .. 
+func NewCodingLanguage(name string) NamedCriterion {
+	return NewNamedCriterion(name)
+} 
 
-// CodingLanguages ..
-type CodingLanguages []CodingLanguage
-
-// HasCodingLanguage ..
-func (c CodingLanguages) HasCodingLanguage(targetLanguage CodingLanguage) bool {
-	for _, language := range c {
-		if language == targetLanguage {
-			return true
-		}
-	}
-	return false
-}
-
-// Match ..
-func (c CodingLanguages) Match(targetLanguages CodingLanguages) CodingLanguages {
-	matches := CodingLanguages{}
-	for _, language := range targetLanguages {
-		if c.HasCodingLanguage(language){
-			matches = append(matches, language)
-		}
-	}
-	return matches
+// NewCodingLanguages ..
+func NewCodingLanguages(languages ...Criterion) NamedCriteria {
+	return NewNamedCriteria(languages...)
```

Comment:
Shouldn't this create newCodingLanguage?

It should also probably take a list of strings and just generate the criterion. There's no need to have a concrete criterion outside the context of the list of criteria. 
Or am I missing something?